ci/jenkins/utils.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function check_and_upgrade_golang() {
+    export GOROOT=/usr/local/go
+    export PATH=${GOROOT}/bin:$PATH
+    echo "====== Checking installed Golang version ======"
+    golang_minor_version=$(go mod edit -json | jq -r .Go)
+    antrea_golang_version="go${golang_minor_version}"
+    installed_version=$(/usr/local/go-${golang_minor_version}/bin/go version | grep -oP "go\d+\.\d+(\.\d+)?") || true
+    # Since the antrea go.mod file does not contain the golang patch number, we will query go.dev
+    # and use the latest patch number.
+    curl -s https://go.dev/dl/\?mode\=json\&include\=all | jq -c -r '.[]' | while read i; do
```

Comment:
It's quite a long list, is it possible to make it shorter before process it? Considering we only use recent two or three versions at most, we may excludes those old versions.

Reply:
I have tried several ways to solve this, but if I removed `include\=all`, the response will only contain the latest release with different architects and OS. I'm not sure if there's some way to get the `include` argument list.

Reply:
Unfortunately, the server doesn't support any option to do server-side filtering. It's possible they don't want to do too much computation or they want to leverage caching.
However, do we see any impact on our side? I think this code should still execute pretty fast.
Maybe there is a way to improve the `jq` filtering here. In the past I used:
```
jq --arg version_prefix "go${GO_VERSION}." -r '. | map(select(. | .version | startswith($version_prefix))) | first
```
But I don't think it will make too much of a difference

Reply:
Thanks @hjiajing @antoninbas ,I have no strong opinion for this change. Let's keep as it is. Thanks.